### PR TITLE
[clang importer] The default apinotes path can be in the Platform's S…

### PR DESCRIPTION
**Pull request by hyp**
…DK instead of runtime resource dir

Swift platforms like Android (on a windows host) uses an SDK that's split from compiler's resource directory, and the apinote files like std.apinotes and posix_filesystem.apinotes are located in the SDK. This change thus makes the clang importer check to see if the api notes are located in the runtime resource directory, or in the SDK.

### CODE REVIEW

**Comment by hyp** [comment]
will add a test too.

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -751,10 +751,21 @@ void importer::getNormalInvocationArguments(
...
                               languageVersion.asAPINotesVersionString());
-  invocationArgStrs.push_back("-iapinotes-modules");
-  invocationArgStrs.push_back((llvm::Twine(searchPathOpts.RuntimeResourcePath) +
-                               llvm::sys::path::get_separator() +
-                               "apinotes").str());
+  std::string apiNotesPath = (llvm::Twine(searchPathOpts.RuntimeResourcePath) +
+                              llvm::sys::path::get_separator() +
+                              "apinotes").str();
+  auto vfs = ctx.SourceMgr.getFileSystem();
+  if (!vfs->exists(apiNotesPath)) {
```
Comment (compnerd) [inline lib/ClangImporter/ClangImporter.cpp:758]: If the path exists, I think that is okay, and we should just be doing a second additional path.

**Comment by finagolfin** [comment]
Seems the same as #75635?

**Comment by hyp** [comment]
closing in favor of @compnerd 's PR

**Pull request closed without merging**